core/pyos.py

```diff
@@ -264,121 +263,26 @@ def InputAvailable(fd):
 
 UNTRAPPED_SIGWINCH = -1
 
-class SigwinchHandler(object):
-  """Wrapper to call user handler."""
 
-  def __init__(self, display, sig_state):
-    # type: (_IDisplay, SignalState) -> None
-    self.display = display
-    self.sig_state = sig_state
-    self.user_handler = None  # type: _TrapHandler
+class SignalHandler(object):
+    def Run(self, sig_num):
+        # type: (int) -> None
+        raise NotImplementedError()
 
-  def __call__(self, sig_num, unused_frame):
-    # type: (int, Any) -> None
-    """For Python's signal module."""
-
-    # SENTINEL for UNTRAPPED SIGWINCH.  If it's trapped, self.user_handler
-    # will overwrite it with signal.SIGWINCH.
-    self.sig_state.last_sig_num = UNTRAPPED_SIGWINCH
-
-    self.display.OnWindowChange()
-    if self.user_handler:
-      self.user_handler(sig_num, unused_frame)
-
-
-def SignalState_AfterForkingChild():
-  # type: () -> None
-  """Not a member of SignalState since we didn't do dependency injection."""
-
-  # Note: this happens in BOTH interactive and non-interactive shells.
-  # We technically don't need to do most of it in non-interactive, since we
-  # did not change state in InitInteractiveShell().
-
-  # Python sets SIGPIPE handler to SIG_IGN by default.  Child processes
-  # shouldn't have this.
-  # https://docs.python.org/2/library/signal.html
-  # See Python/pythonrun.c.
-  signal.signal(signal.SIGPIPE, signal.SIG_DFL)
-
-  # Respond to Ctrl-\ (core dump)
-  signal.signal(signal.SIGQUIT, signal.SIG_DFL)
+    # XXX: We could have done something like `lambda sig, unused: handler.Run(sig)`
+    # in Sigaction(), but that doesn't type check because lambda returns Any and
+    # signal.signal() expects a Callable that returns None (and for some reason
+    # we can't provide type hints for lambdas)
```

Comment:
Hm one idea is to just disable type checks inside the body, which I think is OK  since pyos.py doesn't need to translate.  It is basically an interface shim.

Anything that must translate needs strict types of course

Reply:
I'm not _too_ torn up about not using `lambda` here, but it would help make it more obvious that this and the `SignalHandler` in `leaky_core.h` are the same thing. So, disabling the type check here is probably a net gain